src/lib/eina/eina_debug.c

```diff
@@ -597,37 +597,15 @@ _monitor(void *_data)
 static void
 _thread_start(Eina_Debug_Session *session)
 {
-#ifndef _WIN32
-   pthread_t monitor_thread;
-   int err;
-   sigset_t oldset, newset;
-
-   sigemptyset(&newset);
-   sigaddset(&newset, SIGPIPE);
-   sigaddset(&newset, SIGALRM);
-   sigaddset(&newset, SIGCHLD);
-   sigaddset(&newset, SIGUSR1);
-   sigaddset(&newset, SIGUSR2);
-   sigaddset(&newset, SIGHUP);
-   sigaddset(&newset, SIGQUIT);
-   sigaddset(&newset, SIGINT);
-   sigaddset(&newset, SIGTERM);
-#ifdef SIGPWR
-   sigaddset(&newset, SIGPWR);
-#endif
-   pthread_sigmask(SIG_BLOCK, &newset, &oldset);
+   Eina_Thread monitor_thread;
 
-   err = pthread_create(&monitor_thread, NULL, _monitor, session);
+   Eina_Bool err = eina_thread_create(&monitor_thread,EINA_THREAD_BACKGROUND,-1,(Eina_Thread_Cb)_monitor,session);
 
-   pthread_sigmask(SIG_SETMASK, &oldset, NULL);
-   if (err != 0)
+   if (!err)
```

Comment:
The return values of `pthread_create()` and `eina_thread_create()` differs?
Is it intentional changing from `(err != 0)` to `(!err)`?

Reply:
Hi @joaoantoniocardoso according to [efl thread documentation](https://www.enlightenment.org/develop/legacy/api/c/start#group__Eina__Thread__Group.html), the return value of eina_thread_create() differs from pthread_create().

if (! err) is itentionally for adapt the new code